server/poll/metadata.go

```diff
@@ -2,18 +2,20 @@ package poll
 
 // Metadata stores personalized metadata of a poll.
 type Metadata struct {
```

Comment:
0/5 thought, which it out of scope.: It feels like `Metadata` and `Poll` are very similar. Maybe they can be unified at some point.

Reply:
I thought the same thing, and I tried to add `poll.Settings` field to `Metadata`. But the only data types that can be handled between the server and webapp are those specified by gob [here](https://github.com/mattermost/mattermost-server/blob/e4fb5791b0ccfe22d2e2c8bdb7df5a410675f0a7/plugin/client_rpc.go#L95), so specifying the `poll.Settings` field resulted in an encoding error. As a results, we may not be able to use `Poll` type as `Metadata`.